apis/iam/v1alpha1/referencers.go

```diff
@@ -51,23 +74,39 @@ func ServiceAccountMemberName() reference.ExtractValueFn {
 	}
 }
 
-// ResolveReferences of this ServiceAccountPolicy
-func (in *ServiceAccountPolicy) ResolveReferences(ctx context.Context, c client.Reader) error {
-	r := reference.NewAPIResolver(c, in)
-
+func (sar *ServiceAccountReferer) resolveReferences(ctx context.Context, resolver *reference.APIResolver) error {
 	// Resolve spec.forProvider.serviceAccount
-	rsp, err := r.Resolve(ctx, reference.ResolutionRequest{
-		CurrentValue: reference.FromPtrValue(in.Spec.ForProvider.ServiceAccount),
-		Reference:    in.Spec.ForProvider.ServiceAccountRef,
-		Selector:     in.Spec.ForProvider.ServiceAccountSelector,
+	rsp, err := resolver.Resolve(ctx, reference.ResolutionRequest{
+		CurrentValue: reference.FromPtrValue(sar.ServiceAccount),
+		Reference:    sar.ServiceAccountRef,
+		Selector:     sar.ServiceAccountSelector,
 		To:           reference.To{Managed: &ServiceAccount{}, List: &ServiceAccountList{}},
 		Extract:      ServiceAccountRRN(),
 	})
+
 	if err != nil {
-		return errors.Wrap(err, "spec.forProvider.cryptoKey")
+		return errors.Wrap(err, fmt.Sprintf("could not resolve ServiceAccount reference: %v", *sar))
```

Comment:
`sar` is of type `*ServiceAccountReferer`. What's going to be printed here? I think we could possibly return the error without wrapping since it's wrapped in `ResolveReferences` with the path.